### ISSUE TITLE
LR-Dropout always causes OOM error when training with multi-GPU setup.

### ISSUE THREAD

**Issue by GravitationalLensing** [iperov/DeepFaceLab#851]
## Expected behaviour:
After training a DF-UD model on dual 1070s at batch size 22 for some 300k+ iterations, the time came to enable learning rate dropout. Per the description for the function I did expect an increase in memory usage and was prepared to drop the batch size somewhat as needed.

## Actual behaviour:
Despite Dropping the batch size low as 4 (from 22) the OOM errors persisted and I was not able to train with LR-Dropout enabled on both GPUs. On a whim I tried training at BS11 on a single GPU and oddly enough, it worked without any issue.

This behaviour is consistent regardless of whether LR-Dropout is set to CPU or GPU.

## Steps to reproduce:
(Tested on latest Windows release as of 2020/08/05)
- Train a model as normal to the point it is ready for LR-Dropout
- Enable LR-Dropout and continue training the existing model with both GPUs.
- Observe as it crashes after a few iterations with a output along the lines of:

`2020-08-05 16:00:40.623215: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623215: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623215: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623247: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623251: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623221: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623251: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623253: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623216: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623246: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623252: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623253: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623220: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623217: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623216: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623221: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623261: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623247: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623260: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623221: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623272: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623272: E tensorflow/stream_executor/cuda/cuda_event.cc:48] Error polling for event status: failed to query event: CUDA_ERROR_OUT_OF_MEMORY: out of memory
2020-08-05 16:00:40.623277: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.623282: E tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:740] failed to record completion event; therefore, failed to create inter-stream dependency
2020-08-05 16:00:40.627669: E tensorflow/stream_executor/stream.cc:328] Error recording event in stream: error recording CUDA event on stream 0000015D187467D0: CUDA_ERROR_OUT_OF_MEMORY: out of memory; not marking stream as bad, as the Event object may be at fault. Monitor for further errors.
2020-08-05 16:00:40.634641: F tensorflow/core/common_runtime/gpu/gpu_event_mgr.cc:274] Unexpected Event status: 1
2020-08-05 16:00:40.635982: E tensorflow/stream_executor/cuda/cuda_event.cc:48] Error polling for event status: failed to query event: CUDA_ERROR_OUT_OF_MEMORY: out of memory`

- Repeat with single GPU and observe as the model will without issue train with LR-Dropout despite having a higher batchsize than the size that yielded a OOM error when running with dual GPUs.

## System Specifications:
OS: Windows 10 Pro (v2004)
CPU: R9 3900x
GPU(s): 2 x Nvidia GTX 1070
RAM: 32GB DDR4

##TLDR:
OOM error when training with LR-Dropout at batch size low as 4 on dual 1070s, no OOM error training with it enabled at BS11 on a single 1070.

**Comment by ghost** [comment]
Issue solved / already answered (or it seems like user error), please close it.
